From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Max Filippov <jcmvbkbc@gmail.com>
Date: Sat, 4 Jun 2022 16:32:22 -0700
Subject: [PATCH 23/73] xtensa: xip_kc705_nommu: add defconfig and DTS

Signed-off-by: Max Filippov <jcmvbkbc@gmail.com>
(cherry picked from commit 8024fa0d4ed4183837cee3ea9a73ac3fc4977cf2)
---
 arch/xtensa/boot/dts/xip_kc705.dts            |  36 ++++++
 arch/xtensa/boot/dts/xip_kc705_nommu.dts      |  14 +++
 arch/xtensa/configs/xip_kc705_nommu_defconfig | 117 ++++++++++++++++++
 3 files changed, 167 insertions(+)
 create mode 100644 arch/xtensa/boot/dts/xip_kc705.dts
 create mode 100644 arch/xtensa/boot/dts/xip_kc705_nommu.dts
 create mode 100644 arch/xtensa/configs/xip_kc705_nommu_defconfig

diff --git a/arch/xtensa/boot/dts/xip_kc705.dts b/arch/xtensa/boot/dts/xip_kc705.dts
new file mode 100644
index 0000000000000..3ee7193d3227c
--- /dev/null
+++ b/arch/xtensa/boot/dts/xip_kc705.dts
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+/include/ "kc705.dts"
+
+/ {
+	compatible = "cdns,xtensa-kc705";
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xfd050020,115200n8 console=ttyS0,115200n8 debug rw root=mtd:rootfs";
+	};
+	memory@0 {
+		device_type = "memory";
+		reg = <0x20000000 0x20000000>;
+	};
+	soc {
+		flash: flash@00000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "cfi-flash";
+			reg = <0x00000000 0x08000000>;
+			bank-width = <2>;
+			device-width = <2>;
+			partition@0 {
+				label = "rootfs";
+				reg = <0x00000000 0x01000000>;
+			};
+			partition@1000000 {
+				label = "data";
+				reg = <0x01000000 0x05000000>;
+			};
+			partition@6000000 {
+				label = "XIP kernel image";
+				reg = <0x06000000 0x02000000>;
+			};
+		};
+	};
+};
diff --git a/arch/xtensa/boot/dts/xip_kc705_nommu.dts b/arch/xtensa/boot/dts/xip_kc705_nommu.dts
new file mode 100644
index 0000000000000..269b0defae38a
--- /dev/null
+++ b/arch/xtensa/boot/dts/xip_kc705_nommu.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+/include/ "kc705_nommu.dts"
+
+/ {
+	compatible = "cdns,xtensa-kc705";
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0x9d050020,115200n8 console=ttyS0,115200n8 debug rw root=mtd:data init=/bin/sh ip=dhcp";
+	};
+	memory@0 {
+		device_type = "memory";
+		reg = <0x60000000 0x10000000>;
+	};
+};
diff --git a/arch/xtensa/configs/xip_kc705_nommu_defconfig b/arch/xtensa/configs/xip_kc705_nommu_defconfig
new file mode 100644
index 0000000000000..8e8531af42be1
--- /dev/null
+++ b/arch/xtensa/configs/xip_kc705_nommu_defconfig
@@ -0,0 +1,117 @@
+CONFIG_NO_HZ_IDLE=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_PREEMPT=y
+CONFIG_CC_OPTIMIZE_FOR_SIZE=y
+# CONFIG_MULTIUSER is not set
+# CONFIG_FHANDLE is not set
+# CONFIG_BASE_FULL is not set
+# CONFIG_EPOLL is not set
+# CONFIG_SIGNALFD is not set
+# CONFIG_TIMERFD is not set
+# CONFIG_EVENTFD is not set
+# CONFIG_AIO is not set
+# CONFIG_IO_URING is not set
+# CONFIG_ADVISE_SYSCALLS is not set
+# CONFIG_MEMBARRIER is not set
+CONFIG_EMBEDDED=y
+# CONFIG_VM_EVENT_COUNTERS is not set
+CONFIG_SLOB=y
+CONFIG_XTENSA_VARIANT_CUSTOM=y
+CONFIG_XTENSA_VARIANT_CUSTOM_NAME="de212"
+# CONFIG_XTENSA_VARIANT_MMU is not set
+CONFIG_XTENSA_PLATFORM_XTFPGA=y
+CONFIG_USE_OF=y
+CONFIG_BUILTIN_DTB_NAME="xip_kc705_nommu"
+# CONFIG_PARSE_BOOTPARAM is not set
+CONFIG_XIP_KERNEL=y
+CONFIG_MEMMAP_CACHEATTR=0x2222442f
+CONFIG_KERNEL_LOAD_ADDRESS=0x50000000
+CONFIG_VECTORS_ADDR=0x40000000
+CONFIG_XIP_DATA_ADDR=0x60000000
+# CONFIG_STACKPROTECTOR is not set
+# CONFIG_BLOCK is not set
+CONFIG_BINFMT_FLAT=y
+# CONFIG_COREDUMP is not set
+CONFIG_NET=y
+CONFIG_INET=y
+CONFIG_IP_PNP=y
+CONFIG_IP_PNP_DHCP=y
+# CONFIG_INET_DIAG is not set
+# CONFIG_IPV6 is not set
+# CONFIG_WIRELESS is not set
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+# CONFIG_STANDALONE is not set
+# CONFIG_PREVENT_FIRMWARE_BUILD is not set
+# CONFIG_FW_LOADER is not set
+# CONFIG_ALLOW_DEV_COREDUMP is not set
+CONFIG_MTD=y
+CONFIG_MTD_CFI=y
+CONFIG_MTD_CFI_INTELEXT=y
+CONFIG_MTD_ROM=y
+CONFIG_MTD_PHYSMAP=y
+CONFIG_MTD_PHYSMAP_OF=y
+CONFIG_NETDEVICES=y
+# CONFIG_NET_CORE is not set
+# CONFIG_NET_VENDOR_ALACRITECH is not set
+# CONFIG_NET_VENDOR_AMAZON is not set
+# CONFIG_NET_VENDOR_AQUANTIA is not set
+# CONFIG_NET_VENDOR_ARC is not set
+# CONFIG_NET_VENDOR_BROADCOM is not set
+# CONFIG_NET_VENDOR_CADENCE is not set
+# CONFIG_NET_VENDOR_CAVIUM is not set
+# CONFIG_NET_VENDOR_CORTINA is not set
+# CONFIG_NET_VENDOR_EZCHIP is not set
+# CONFIG_NET_VENDOR_GOOGLE is not set
+# CONFIG_NET_VENDOR_HUAWEI is not set
+# CONFIG_NET_VENDOR_INTEL is not set
+# CONFIG_NET_VENDOR_MARVELL is not set
+# CONFIG_NET_VENDOR_MICREL is not set
+# CONFIG_NET_VENDOR_MICROCHIP is not set
+# CONFIG_NET_VENDOR_MICROSEMI is not set
+# CONFIG_NET_VENDOR_NI is not set
+# CONFIG_NET_VENDOR_NATSEMI is not set
+# CONFIG_NET_VENDOR_NETRONOME is not set
+# CONFIG_NET_VENDOR_PENSANDO is not set
+# CONFIG_NET_VENDOR_QUALCOMM is not set
+# CONFIG_NET_VENDOR_RENESAS is not set
+# CONFIG_NET_VENDOR_ROCKER is not set
+# CONFIG_NET_VENDOR_SAMSUNG is not set
+# CONFIG_NET_VENDOR_SEEQ is not set
+# CONFIG_NET_VENDOR_SOLARFLARE is not set
+# CONFIG_NET_VENDOR_SMSC is not set
+# CONFIG_NET_VENDOR_SOCIONEXT is not set
+# CONFIG_NET_VENDOR_STMICRO is not set
+# CONFIG_NET_VENDOR_SYNOPSYS is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NET_VENDOR_XILINX is not set
+CONFIG_NATIONAL_PHY=y
+# CONFIG_WLAN is not set
+# CONFIG_INPUT is not set
+# CONFIG_SERIO is not set
+# CONFIG_VT is not set
+CONFIG_SERIAL_8250=y
+# CONFIG_SERIAL_8250_DEPRECATED_OPTIONS is not set
+CONFIG_SERIAL_8250_CONSOLE=y
+CONFIG_SERIAL_OF_PLATFORM=y
+# CONFIG_HW_RANDOM is not set
+# CONFIG_HWMON is not set
+# CONFIG_USB_SUPPORT is not set
+# CONFIG_VIRTIO_MENU is not set
+# CONFIG_NVMEM is not set
+# CONFIG_FILE_LOCKING is not set
+# CONFIG_DNOTIFY is not set
+# CONFIG_INOTIFY_USER is not set
+# CONFIG_SYSFS is not set
+CONFIG_CRAMFS=y
+CONFIG_LSM="yama,loadpin,safesetid,integrity"
+CONFIG_PRINTK_TIME=y
+# CONFIG_SYMBOLIC_ERRNAME is not set
+# CONFIG_DEBUG_MISC is not set
+# CONFIG_FRAME_POINTER is not set
+# CONFIG_DEBUG_PREEMPT is not set
+# CONFIG_FTRACE is not set
+# CONFIG_LD_NO_RELAX is not set
+# CONFIG_S32C1I_SELFTEST is not set
+# CONFIG_RUNTIME_TESTING_MENU is not set
